password_change_restrict/models/res_users.py

```diff
@@ -0,0 +1,48 @@
+# -*- coding: utf-8 -*-
+# License and authorship information in:
+# __openerp__.py file at the root folder of this module.
+
+from openerp import api, exceptions, models, SUPERUSER_ID, _
+
+
+class ResUsers(models.Model):
+    _inherit = 'res.users'
+
+    SELF_WRITEABLE_FIELDS = ['signature', 'action_id', 'company_id', 'email',
+                             'name', 'image', 'image_medium', 'image_small',
+                             'lang', 'tz']
+
+    def _can_change_password(self):
+        group_ok = 'base.group_erp_manager'
+        user = self.env['res.users'].browse(self.env.uid)
```

Comment:
the function is decorated with `@tools.ormcache(skiparg=2)` which seems weird to me (I'd have expected `skiparg=1`), so I'd guess this is a caching issue.

I personally use `self.env.ref('group.xmlid') in self.env.user.groups_id`